core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -0,0 +1,5534 @@
+      }
+    } else {
+
+      return object;
+
+    }
+  }
+
+  @Override
+  public String getCompatibleString( Object object ) throws KettleValueException {
+    try {
+      String string;
+
+      switch ( type ) {
+        case TYPE_DATE:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              string = convertDateToCompatibleString( (Date) object );
+              break;
+            case STORAGE_TYPE_BINARY_STRING:
+              string = convertDateToCompatibleString( (Date) convertBinaryStringToNativeType( (byte[]) object ) );
+              break;
+            case STORAGE_TYPE_INDEXED:
+              if ( object == null ) {
+                string = null;
+              } else {
+                string = convertDateToCompatibleString( (Date) index[( (Integer) object ).intValue()] );
+              }
+              break;
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+          break;
+
+        case TYPE_NUMBER:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              string = convertNumberToCompatibleString( (Double) object );
+              break;
+            case STORAGE_TYPE_BINARY_STRING:
+              string = convertNumberToCompatibleString( (Double) convertBinaryStringToNativeType( (byte[]) object ) );
+              break;
+            case STORAGE_TYPE_INDEXED:
+              string =
+                  object == null ? null : convertNumberToCompatibleString( (Double) index[( (Integer) object )
+                      .intValue()] );
+              break;
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+          break;
+
+        case TYPE_INTEGER:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              string = convertIntegerToCompatibleString( (Long) object );
+              break;
+            case STORAGE_TYPE_BINARY_STRING:
+              try {
+                string = convertIntegerToCompatibleString( (Long) convertBinaryStringToNativeType( (byte[]) object ) );
+              } catch ( ClassCastException e ) {
+                string = convertIntegerToCompatibleString( (Long) object );
+              }
+              break;
+            case STORAGE_TYPE_INDEXED:
+              string =
+                  object == null ? null : convertIntegerToCompatibleString( (Long) index[( (Integer) object )
+                      .intValue()] );
+              break;
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+          break;
+
+        default:
+          return getString( object );
+      }
+
+      return string;
+    } catch ( ClassCastException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)